From 31344530ca760de1fc5659f3d7c4841a456ee5fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20=C5=81obacz?= <p.lobacz@welotec.com>
Date: Wed, 26 Jun 2024 16:05:43 +0200
Subject: [PATCH] Change panic strategy to unwind

---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 84f7534..f2630af 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,10 +54,10 @@ members = [
 # Don't let panics from callbacks cross FFI boundary
 
 [profile.dev]
-panic = "abort"
+panic = "unwind"
 
 [profile.release]
-panic = "abort"
+panic = "unwind"
 # Release builds will have full symbols. The packaging phase will strip symbols from binaries and
 # make them available in a separate package.
 debug = 2
-- 
2.34.1
