From 66cfcfb408354e658c889d04f3e936472f7bf494 Mon Sep 17 00:00:00 2001
From: gaya <33068020+ggjjj@users.noreply.github.com>
Date: Wed, 26 Oct 2022 03:27:42 +0000
Subject: [PATCH] Change abort startegy to unwind from abort

---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index c916bec..7eaaa49 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,7 +54,7 @@ members = [
 # Don't let panics from callbacks cross FFI boundary
 
 [profile.dev]
-panic = "abort"
+panic = "unwind"
 
 [profile.release]
-panic = "abort"
+panic = "unwind"
-- 
2.37.0

