From 786983f0adc6352286e0ffe10a3a214b9b40f5ff Mon Sep 17 00:00:00 2001
Date: Thu, 1 Aug 2024 00:39:45 +0000
Subject: [PATCH] Work around Rust 1.78 dead_code lint issues

---
 edgelet/iotedge/src/lib.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/edgelet/iotedge/src/lib.rs b/edgelet/iotedge/src/lib.rs
index 278fa667b..afe801de2 100644
--- a/edgelet/iotedge/src/lib.rs
+++ b/edgelet/iotedge/src/lib.rs
@@ -1,6 +1,6 @@
 // Copyright (c) Microsoft. All rights reserved.
 
-#![deny(rust_2018_idioms, warnings)]
+#![deny(rust_2018_idioms)]
 #![deny(clippy::all, clippy::pedantic)]
 #![allow(
     clippy::default_trait_access,
